x/zoneconcierge/abci.go

```diff
@@ -23,13 +23,19 @@ func BeginBlocker(ctx context.Context, k keeper.Keeper) error {
 func EndBlocker(ctx context.Context, k keeper.Keeper) ([]abci.ValidatorUpdate, error) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
 
+	// Build a map for O(1) channel lookups
+	consumerChannelMap, err := k.GetConsumerChannelMap(ctx)
+	if err != nil {
+		handleBroadcastError(ctx, k, "BuildConsumerChannelMap", err)
+	}
+
 	// Handle BTC headers broadcast with structured error handling
-	if err := k.BroadcastBTCHeaders(ctx); err != nil {
+	if err := k.BroadcastBTCHeaders(ctx, consumerChannelMap); err != nil {
```

Comment:
for future improvements I see that we are getting a slice and sorting to iterate over the consumers (which is correct to avoid non-determinism)

```go
	consumerIDs := make([]string, 0, len(consumerIBCPacketMap))
	for consumerID := range consumerIBCPacketMap {
		consumerIDs = append(consumerIDs, consumerID)
	}
	sort.Strings(consumerIDs)
```

We could also do that once and send to both funcs